File: app/src/main/java/wxm/dofcalculator/db/DeviceDBUtility.java
package wxm.dofcalculator.db;

import com.j256.ormlite.dao.RuntimeExceptionDao;

import org.greenrobot.eventbus.EventBus;

import java.util.List;

import wxm.androidutil.DBHelper.DBUtilityBase;
import wxm.dofcalculator.define.DeviceItem;
import wxm.dofcalculator.utility.ContextUtil;

/**
 * device db utility
 * Created by WangXM on 2017/03/13.
 */
public class DeviceDBUtility extends DBUtilityBase<DeviceItem, Integer> {
    public DeviceDBUtility()  {
        super();
    }

    public int getCount()   {
        return getAllData().size();
    }

    @Override
    protected RuntimeExceptionDao<DeviceItem, Integer> getDBHelper() {
        return ContextUtil.getDBHelper().getREDDevice();
    }


    @Override
    protected void onDataModify(List<Integer> md) {
        EventBus.getDefault().post(new DBDataChangeEvent());
    }

    @Override
    protected void onDataCreate(List<Integer> cd) {
        EventBus.getDefault().post(new DBDataChangeEvent());
    }

    @Override
    protected void onDataRemove(List<Integer> dd) {
        EventBus.getDefault().post(new DBDataChangeEvent());
    }
}

File: app/src/main/java/wxm/dofcalculator/db/DeviceDBUtility.kt
package wxm.dofcalculator.db

import com.j256.ormlite.dao.RuntimeExceptionDao

import org.greenrobot.eventbus.EventBus

import wxm.androidutil.DBHelper.DBUtilityBase
import wxm.dofcalculator.define.DeviceItem
import wxm.dofcalculator.utility.ContextUtil

/**
 * device db utility
 * Created by WangXM on 2017/03/13.
 */
class DeviceDBUtility : DBUtilityBase<DeviceItem, Int>() {

    val count: Int
        get() = allData.size

    override fun getDBHelper(): RuntimeExceptionDao<DeviceItem, Int> {
        return ContextUtil.dbHelper.redDevice
    }


    override fun onDataModify(md: List<Int>) {
        EventBus.getDefault().post(DBDataChangeEvent())
    }

    override fun onDataCreate(cd: List<Int>) {
        EventBus.getDefault().post(DBDataChangeEvent())
    }

    override fun onDataRemove(dd: List<Int>) {
        EventBus.getDefault().post(DBDataChangeEvent())
    }
}
